### ISSUE TITLE
Suggest to export symbols if linking with shared_library and *.lib is not found

### ISSUE THREAD

**Issue by sarum9in** [mesonbuild/meson#3187]
This is a proposal to improve error messages for visual studio and assist users. This feature is low priority but might be pretty valuable for any new developers.

`LINK : fatal error LNK1104: cannot open file 'shared.lib'` is not very helpful error message, maybe we could detect this case and print better error.

**Comment by jpakkane** [comment]
At least for the Ninja backend there is nothing we can really do. When the error gets printed, the only executables that are running are Ninja and the linker. There are no Meson processes around to do something unless we want to run every command via a wrapper which seems like a complexity and perf problem.

**Comment by sarum9in** [comment]
That's true. Let's leave this open for now maybe someone comes up with some reasonable way to improve this. I understand it is non-trivial but I still believe it is worth to have if can be done in a reasonable way.
